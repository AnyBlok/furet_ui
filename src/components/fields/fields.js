export const fields = {
  list: {},
  form: {},
  thumbnail: {},
  kanban: {},
};
export default {
    fields,
}
