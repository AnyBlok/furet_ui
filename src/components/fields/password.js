/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import {defineComponent} from '../factory'
import {fields} from './fields';
import { listTemplate } from './common';
import _ from 'underscore';
import {safe_eval} from './common';


defineComponent('furet-ui-list-field-password', {
  template: listTemplate,
  extend: ['furet-ui-list-field-password'],
  prototype: {
    computed: {
      value () {
        return _.map(this.data[this.condif.name] || '', () => '*').join('');
      },
    },
  },
})
fields.list.password = 'furet-ui-list-field-password'

// export const FieldThumbnailPassword = Vue.component('furet-ui-thumbnail-field-password', {
//     mixins: [ThumbnailMixin],
//     computed: {
//         value () {
//             return _.map(this.data && this.data[this.name] || '', a => '*').join('');
//         },
//     },
// })

defineComponent('furet-ui-form-field-password', {
  template: `
    <furet-ui-form-field-common-tooltip-field
      v-bind:resource="resource"
      v-bind:data="data"
      v-bind:config="config"
    >
      <span v-if="isReadonly">{{ readonly_value }}</span>
      <b-input 
        v-else
        v-bind:value="value" 
        v-on:input="updateValue"
        v-bind:maxlength="config.maxlength"
        v-bind:placeholder="config.placeholder"
        icon-pack="fa"
        v-bind:icon="config.icon"
        type="password"
        v-bind:password-reveal="reveal"
      />
    </furet-ui-form-field-common-tooltip-field>
  `,
  extend: ['furet-ui-form-field-common'],
  prototype: {
    computed: {
      readonly_value () {
        return _.map(this.data[this.config.name] || '', () => '*').join('');
      },
      reveal () {
        if (this.isReadonly) return false;
        return safe_eval(this.config.reveal, this.data || {}, this.resource.selectors);
      },
    },
  }
})
fields.form.password = 'furet-ui-form-field-password'
