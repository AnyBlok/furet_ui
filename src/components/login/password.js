import axios from "axios";
import { defineComponent } from "../factory";
import { loginPages } from "./login-pages";

defineComponent("furet-ui-login-password", {
  template: `
    <div class="container has-text-centered">
      <form v-on:submit="logIn">
         <furet-ui-page-errors v-bind:errors="errors" />
         <b-field v-bind:label="$t('components.login.username')">
           <b-input v-model="username" expanded></b-input>
         </b-field>
         <b-field v-bind:label="$t('components.login.password')">
           <b-input v-model="password" type="password" expanded password-reveal></b-input>
         </b-field>
         <br />
         <div class="buttons">
           <b-button 
             type="is-primary" 
             native-type="submit" 
             expanded
             v-bind:disabled="is_not_clickable"
           >
           {{ $t('components.login.button') }}
           </b-button>
         </div>
      </form>
    </div>
  `,
  prototype: {
    data() {
      return {
        username: "",
        password: "",
        errors: [],
      };
    },
    computed: {
      is_not_clickable() {
        if (!this.username) return true;
        if (!this.password) return true;
        return false;
      },
    },
    methods: {
      logIn() {
        if (this.is_not_clickable) return;
        this.errors = [];
        axios
          .post("/furet-ui/login", {
            login: this.username,
            password: this.password,
            redirect: this.$route.query.redirect,
          })
          .then((result) => {
            this.$dispatchAll(result.data);
          })
          .catch((error) => {
            this.errors = [error];
          });
      },
    },
  },
});

loginPages.password = "furet-ui-login-password";
