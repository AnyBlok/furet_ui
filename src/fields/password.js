/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import {FormMixin, ThumbnailMixin, ListMixin} from './common';
import _ from 'underscore';


export const FieldListPassword = Vue.component('furet-ui-list-field-password', {
    mixins: [ListMixin],
    template: `
        <span v-if="isInvisible" />
        <span v-else>{{value}}</span>`,
    computed: {
        value () {
            return _.map(this.row[this.header.name] || '', a => '*').join('');
        },
    },
})


export const FieldThumbnailPassword = Vue.component('furet-ui-thumbnail-field-password', {
    mixins: [ThumbnailMixin],
    template: `
        <div v-if="this.isInvisible" />
        <b-tooltip 
            v-bind:label="getTooltip" 
            v-bind:position="tooltipPosition"
            v-else
        >
            <b-field 
                v-bind:label="this.label"
                v-bind:style="{'width': 'inherit'}"
            >
                <span> {{value}} </span>
            </b-field>
        </b-tooltip>`,
    computed: {
        value () {
            return _.map(this.data && this.data[this.name] || '', a => '*').join('');
        },
    },
})

export const FieldFormPassword = Vue.component('furet-ui-form-field-password', {
    props: ['maxlength', 'placeholder', 'icon', 'reveal'],
    mixins: [FormMixin],
    template: `
        <div v-if="this.isInvisible" />
        <b-tooltip 
            v-bind:label="getTooltip" 
            v-bind:position="tooltipPosition"
            v-bind:style="{'width': '100%'}"
            v-else
        >
            <b-field 
                v-bind:label="this.label"
                v-bind:type="getType"
                v-bind:message="getMessage"
                v-bind:style="{'width': 'inherit'}"
            >
                <span v-if="isReadonly"> {{value}} </span>
                <b-input 
                    v-else 
                    v-bind:value="data" 
                    v-on:change="updateValue"
                    v-bind:maxlength="maxlength"
                    v-bind:placeholder="placeholder"
                    icon-pack="fa"
                    v-bind:icon="icon"
                    type="password"
                    v-bind:password-reveal="reveal"
                >
                </b-input>
            </b-field>
        </b-tooltip>`,
    computed: {
        value () {
            return _.map(this.data, a => '*').join('');
        },
    },
})
