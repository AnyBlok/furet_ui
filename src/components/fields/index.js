import { fields } from './fields';
import './common';
import './boolean';
import './datetime';
import './integer';
import './password';
import './timestamp';
import './selection';
import './string';

import './relationship';

import { defineComponent } from '../factory';

defineComponent('furet-ui-unknown-field', {
  template : `<div>Unknow Field <strong>{{config.type}}</strong></div>`,
  prototype: {
    props: ['config', 'data'],
  }
})

defineComponent('furet-ui-field', {
  template : `
    <component 
      v-bind:is="component" 
      v-bind:resource="resource" 
      v-bind:config="config" 
      v-bind:data="data"
    />
  `,
  prototype: {
    props: ['resource', 'config', 'data'],
    computed: {
      component () {
        return (fields[this.resource.type] || {})[this.config.type] || 'furet-ui-unknown-field'
      },
    },
  }
})
