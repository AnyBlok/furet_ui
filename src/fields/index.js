/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import React from 'react';
import plugin from '../plugin';
import './string';
import './selection';
import './date';
import './integer';
import './float';
import './url';
import './uuid';
import './sequence';
import './password';
import './color';
import './text';

/**
 * Unknown field, used only if the wanted field is not available
**/
plugin.set(['field'], {Unknown: (props) => {
    return <div>{props.value}</div>
}});

/**
 * return the component for viewType and fieldType
**/
export const getField = (viewType, fieldType, attribs, value) => {
    let field = plugin.get(['field', viewType, fieldType]);
    if (!field) {
        field = plugin.get(['field', 'Unknown']);
        console.log('getField', viewType, fieldType, attribs, value)
    }
    const id = 'view-type-' + viewType + '-field_type-' + fieldType + '-field-name-' + (attribs.name || '');
    return React.createElement(field, Object.assign({id}, attribs, {value}));
}

export default {
    getField,
}
