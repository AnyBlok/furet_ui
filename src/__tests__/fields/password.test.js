/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import Vuex from 'vuex';
import Buefy from 'buefy';
Vue.use(Buefy, {defaultIconPack: 'fa',});
import {store} from '../../store';
import {router} from '../../routes';
import {i18n} from '../../i18n';
import {FieldListPassword, FieldThumbnailPassword, FieldFormPassword} from '../../fields/password'

describe('Password list component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
    });
    it('Render with data', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListPassword, {props: {
                row: {fieldname: '#123456'},
                header: {name: 'fieldname'},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render without data', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListPassword, {props: {
                row: {},
                header: {name: 'fieldname'},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is True', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListPassword, {props: {
                row: {fieldname: '#123456'},
                header: {name: 'fieldname', invisible: true},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListPassword, {props: {
                row: {fieldname: '#123456'},
                header: {name: 'fieldname', invisible: false},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is true', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListPassword, {props: {
                row: {fieldname: '#123456', invisible: true},
                header: {name: 'fieldname', invisible: 'fields.invisible'},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListPassword, {props: {
                row: {fieldname: '#123456', invisible: false},
                header: {name: 'fieldname', invisible: 'fields.invisible'},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});

describe('Password Thumbnail component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
    });
    it('Render', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailPassword, {props: {
                data: {fieldname: '#123456'},
                name: 'fieldname',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render without data', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailPassword, {props: {
                data: {},
                name: 'fieldname',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is True', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailPassword, {props: {
                data: {fieldname: '#123456'},
                name: 'fieldname',
                invisible: true,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailPassword, {props: {
                data: {fieldname: '#123456'},
                name: 'fieldname',
                invisible: false,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is true', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailPassword, {props: {
                data: {fieldname: '#123456', invisible: true},
                name: 'fieldname',
                invisible: 'fields.invisible',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailPassword, {props: {
                data: {fieldname: '#123456', invisible: false},
                name: 'fieldname',
                invisible: 'fields.invisible',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render label', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailPassword, {props: {
                data: {fieldname: '#123456'},
                name: 'fieldname',
                label: 'The label',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render tooltip up', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailPassword, {props: {
                data: {fieldname: '#123456'},
                name: 'fieldname',
                tooltip: 'The tooltip',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render tooltip left', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailPassword, {props: {
                data: {fieldname: '#123456'},
                name: 'fieldname',
                tooltip: 'The left tooltip',
                tooltip_position: 'is-left',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});

describe('Password Form component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
    });
    it('Render', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {fieldname: '#123456'}},
                name: 'fieldname',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render without data', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {}},
                name: 'fieldname',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is True', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {fieldname: '#123456'}},
                name: 'fieldname',
                invisible: true,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {fieldname: '#123456'}},
                name: 'fieldname',
                invisible: false,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is true', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {fieldname: '#123456', invisible: true}},
                name: 'fieldname',
                invisible: 'fields.invisible',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {fieldname: '#123456', invisible: false}},
                name: 'fieldname',
                invisible: 'fields.invisible',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render label', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {fieldname: '#123456'}},
                name: 'fieldname',
                label: 'The label',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render tooltip up', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {fieldname: '#123456'}},
                name: 'fieldname',
                tooltip: 'The tooltip',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render tooltip left', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {fieldname: '#123456'}},
                name: 'fieldname',
                tooltip: 'The left tooltip',
                tooltip_position: 'is-left',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render mode readonly by config', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {fieldname: '#123456'}, mode: 'readonly'},
                name: 'fieldname',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render mode readonly param', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {fieldname: '#123456'}},
                name: 'fieldname',
                readonly: true,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render mode readonly conditional param', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {fieldname: '#123456', readonly: 1}},
                name: 'fieldname',
                readonly: 'fields.readonly',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render required', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {fieldname: '#123456'}},
                name: 'fieldname',
                required: true,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render required conditionnal', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {fieldname: '#123456', required: true}},
                name: 'fieldname',
                required: 'fields.required',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render required without value', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {}},
                name: 'fieldname',
                required: true,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render placeholder', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {}},
                name: 'fieldname',
                placeholder: 'The placeholder'
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render icon', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {}},
                name: 'fieldname',
                icon: 'user'
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render maxlenght (OK)', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {fieldname: 'password'}},
                name: 'fieldname',
                maxlength: 64,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render maxlenght (KO)', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {fieldname: 'password'}},
                name: 'fieldname',
                maxlength: 4,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render reveal', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormPassword, {props: {
                config: {data: {fieldname: 'password'}},
                name: 'fieldname',
                reveal: true,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});
