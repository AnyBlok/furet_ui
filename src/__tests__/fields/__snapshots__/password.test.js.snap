// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Form component Render 1`] = `<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <div class="control is-clearfix"><input type="password" autocomplete="on" value="#123456" class="input"> <!----> <!----> <!----></div> <!----></div></span>`;

exports[`Password Form component Render icon 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <div class="control has-icons-left is-clearfix"><input type="password" autocomplete="on" value="" class="input"> <span class="icon is-left"><i class="fa fa-user"></i></span>
<!---->
<!---->
</div>
<!---->
</div>
</span>
`;

exports[`Password Form component Render invisible condition is false 1`] = `<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <div class="control is-clearfix"><input type="password" autocomplete="on" value="#123456" class="input"> <!----> <!----> <!----></div> <!----></div></span>`;

exports[`Password Form component Render invisible condition is true 1`] = `<div data-server-rendered="true"></div>`;

exports[`Password Form component Render invisible is True 1`] = `<div data-server-rendered="true"></div>`;

exports[`Password Form component Render invisible is false 1`] = `<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <div class="control is-clearfix"><input type="password" autocomplete="on" value="#123456" class="input"> <!----> <!----> <!----></div> <!----></div></span>`;

exports[`Password Form component Render label 1`] = `<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">The label</label> <div class="control is-clearfix"><input type="password" autocomplete="on" value="#123456" class="input"> <!----> <!----> <!----></div> <!----></div></span>`;

exports[`Password Form component Render maxlenght (KO) 1`] = `<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <div class="control is-clearfix"><input type="password" maxlength="4" autocomplete="on" value="password" class="input"> <!----> <!----> <small class="help counter">8 / 4</small></div> <!----></div></span>`;

exports[`Password Form component Render maxlenght (OK) 1`] = `<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <div class="control is-clearfix"><input type="password" maxlength="64" autocomplete="on" value="password" class="input"> <!----> <!----> <small class="help counter">8 / 64</small></div> <!----></div></span>`;

exports[`Password Form component Render mode readonly by config 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <span> ******* </span>
<!---->
</div>
</span>
`;

exports[`Password Form component Render mode readonly conditional param 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <span> ******* </span>
<!---->
</div>
</span>
`;

exports[`Password Form component Render mode readonly param 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <span> ******* </span>
<!---->
</div>
</span>
`;

exports[`Password Form component Render placeholder 1`] = `<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <div class="control is-clearfix"><input type="password" placeholder="The placeholder" autocomplete="on" value="" class="input"> <!----> <!----> <!----></div> <!----></div></span>`;

exports[`Password Form component Render required 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <div class="control has-icons-right is-clearfix"><input type="password" autocomplete="on" value="#123456" class="input is-info"> <!----> <span class="icon is-right is-info"><i class="fa fa-info-circle"></i></span>
<!---->
</div>
<!---->
</div>
</span>
`;

exports[`Password Form component Render required conditionnal 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <div class="control has-icons-right is-clearfix"><input type="password" autocomplete="on" value="#123456" class="input is-info"> <!----> <span class="icon is-right is-info"><i class="fa fa-info-circle"></i></span>
<!---->
</div>
<!---->
</div>
</span>
`;

exports[`Password Form component Render required without value 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <div class="control has-icons-right"><input type="password" autocomplete="on" value="" class="input is-danger"> <!----> <span class="icon is-right is-danger"><i class="fa fa-exclamation-circle"></i></span>
<!---->
</div>
<p class="help is-danger">this field is required</p>
</div>
</span>
`;

exports[`Password Form component Render reveal 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <div class="control has-icons-right is-clearfix"><input type="password" autocomplete="on" value="password" class="input"> <!----> <span class="icon is-right is-primary is-clickable"><i class="fa fa-eye"></i></span>
<!---->
</div>
<!---->
</div>
</span>
`;

exports[`Password Form component Render tooltip left 1`] = `<span data-label="The left tooltip" data-server-rendered="true" class="tooltip is-primary is-left is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <div class="control is-clearfix"><input type="password" autocomplete="on" value="#123456" class="input"> <!----> <!----> <!----></div> <!----></div></span>`;

exports[`Password Form component Render tooltip up 1`] = `<span data-label="The tooltip" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <div class="control is-clearfix"><input type="password" autocomplete="on" value="#123456" class="input"> <!----> <!----> <!----></div> <!----></div></span>`;

exports[`Password Form component Render without data 1`] = `<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <div class="control is-clearfix"><input type="password" autocomplete="on" value="" class="input"> <!----> <!----> <!----></div> <!----></div></span>`;

exports[`Password Thumbnail component Render 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!----> <span> ******* </span>
<!---->
</div>
</span>
`;

exports[`Password Thumbnail component Render invisible condition is false 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!----> <span> ******* </span>
<!---->
</div>
</span>
`;

exports[`Password Thumbnail component Render invisible condition is true 1`] = `<div data-server-rendered="true"></div>`;

exports[`Password Thumbnail component Render invisible is True 1`] = `<div data-server-rendered="true"></div>`;

exports[`Password Thumbnail component Render invisible is false 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!----> <span> ******* </span>
<!---->
</div>
</span>
`;

exports[`Password Thumbnail component Render label 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><label class="label">The label</label> <span> ******* </span>
<!---->
</div>
</span>
`;

exports[`Password Thumbnail component Render tooltip left 1`] = `
<span data-label="The left tooltip" data-server-rendered="true" class="tooltip is-primary is-left is-medium"><div class="field" style="width:inherit;"><!----> <span> ******* </span>
<!---->
</div>
</span>
`;

exports[`Password Thumbnail component Render tooltip up 1`] = `
<span data-label="The tooltip" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!----> <span> ******* </span>
<!---->
</div>
</span>
`;

exports[`Password Thumbnail component Render without data 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!----> <span>  </span>
<!---->
</div>
</span>
`;

exports[`Password list component Render invisible condition is false 1`] = `<span data-server-rendered="true">*******</span>`;

exports[`Password list component Render invisible condition is true 1`] = `<span data-server-rendered="true"></span>`;

exports[`Password list component Render invisible is True 1`] = `<span data-server-rendered="true"></span>`;

exports[`Password list component Render invisible is false 1`] = `<span data-server-rendered="true">*******</span>`;

exports[`Password list component Render with data 1`] = `<span data-server-rendered="true">*******</span>`;

exports[`Password list component Render without data 1`] = `<span data-server-rendered="true"></span>`;
